src/StateReaderTaskEither.ts

```diff
@@ -509,22 +509,32 @@ export const chain: <S, R, E, A, B>(
 ) => (ma: StateReaderTaskEither<S, R, E, A>) => StateReaderTaskEither<S, R, E, B> = chainW
 
 /**
- * Composes computations in sequence, using the return value of one computation to determine the next computation and
- * keeping only the result of the first.
+ * Less strict version of [`chainFirst`](#chainFirst).
  *
  * @category Monad
- * @since 2.0.0
+ * @since 2.8.0
  */
-export const chainFirst: <S, R, E, A, B>(
-  f: (a: A) => StateReaderTaskEither<S, R, E, B>
-) => (ma: StateReaderTaskEither<S, R, E, A>) => StateReaderTaskEither<S, R, E, A> = (f) =>
-  chain((a) =>
+export const chainFirstW: <S, R, D, A, B>(
+  f: (a: A) => StateReaderTaskEither<S, R, D, B>
+) => <E>(ma: StateReaderTaskEither<S, R, E, A>) => StateReaderTaskEither<S, R, D | E, A> = (f) =>
```

Comment:
the return type should be `StateReaderTaskEither<S, Q & R, D | E, A>`

Reply:
Dammit. 😅 

8th time lucky?